### ISSUE TITLE
Losing pdf background 

### ISSUE THREAD

**Issue by supersarma** [saber-notes/saber#1066]
### Describe the bug

Hi,
 I am using the pdf import option in order to annotate them. I also add some blank page into the pdf. These files are not huge. Usually les than 5  pages. 
Very often when I rearrange all these pages, I lose the background pdf on some of them.
I can se the circle in the middle of thr page (like buffering).
Also, I see the same thing when I click on options. 
See attached.

![IMG_20231217_220932](https://github.com/saber-notes/saber/assets/55811457/5f8c62f2-da0f-4859-81b1-fbbef817cff5)
![IMG_20231217_220941](https://github.com/saber-notes/saber/assets/55811457/e8236d9f-bf72-4211-a255-6fafc33fb687)

I am not able to reproduce it every time.

### To reproduce

1. Importet pdf file
2. Write on it
3. Add some blank pages to it
4. Rearange pages 
5. Remove the last blank page (if the delete button is grey, move the page ip and then remove it.)
6. Save the file 

### Expected behavior

The background of the pdf file should not disappear. 

### Saber version

0.18.4 (F-Droid)

### Device

- Device: Lenovo P12
- OS : Android 13

**Comment by droghio** [comment]
I have been able to reproduce this on Saber version v0.19.0 on a Amazon Fire 11 Max running FireOS 8.3.2.1 (Android 11).

That said the issue seems intermittent. Sometimes I can follow the above steps and all works as expected other times I follow the steps and it results in a spinner for the PDF background. (This is when importing the same PDF document.)

**Comment by QubaB** [comment]
I have the following steps to reproduce problem (I used debugging of source code to see files on sd card):
1. Import note as pdf file with few pages.
2. Scroll to the last page
3. Save note.
4. Exit notes editor 
 On the sd card are created the following files (my note name is Podstránský (9)):
![image](https://github.com/saber-notes/saber/assets/74395874/ef843629-7a19-4807-afda-4839e91d05d0)
6. Open again note which was previously created.
7. Scroll from last page to first page and draw here some line to enable saving.
8. Save note
  On the sd card are created the following files:
![image](https://github.com/saber-notes/saber/assets/74395874/1fc23d72-a9f4-40c6-8c51-609698e9d825)
  it means that number of assets (is not 1 as on previous save but in my case 9).   
9. Exit notes editor
10. Open note again. Page 1 is OK, but page 2 is not displayed correctly and rotating circle is displayed.

I think that there is some problem with assets during saving (point 8).  But I do not understand source code and studying it takes me too much time.

Tested on source code of v0.19.0

**Comment by mufasachan** [comment]
Hello, anyone knows how to recover the saber note after that?
I guess if you use a "WebDAV service" (e.g. Nextcloud), there might be a recover action or something. In my case the note is local. I created a folder with the same name as my note. Then I renamed the corrupted file "1" and I imported the PDF once again in a note called "2". This is an awkward fix, so I am asking if anyone knows better?

(Thanks for Saber, the app is a gem).

**Comment by 5FeetUnder** [comment]
For me this happens even without writing on it:
1. Import PDF as note
2. Import another PDF inside that note
3. Exit the note

Sometimes it happens directly, sometimes only after I've imported other PDF into other notes.

Not all pages have a missing background.

Just to make sure: missing background means that instead of the page from the PDF it shows only a loading circle, right?
(If not, this should probably be its own issue)

**Comment by n8xm** [comment]
I seem to be running into the same / similar issue. I cannot reproduce it consistently - it seems to happen every once in a while? What happened to me once or twice is the following:

**Expected behavior:** note (with complete PDF as "background") opens as usual, with annotations.
**Actual behavior:** only the pages that have been annotated persist. The other pages have a "circle in the middle of the page" as described by the original bug report.

**Saber version:** `v0.19.2 F-Droid` (19020)
**Device:** Galaxy Tab A with S Pen (SM-P200)
**OS:** Android 11

P.S. This is an amazing app! :pray: If I can figure out how to reproduce it more consistently I will post a follow-up here.

**Comment by n8xm** [comment]
(This was reworked from an earlier comment with steps to consistently reproduce.)
I'm experiencing the same or similar bug.

**Steps to reproduce:**
1. Import note.
2. Choose multi-page PDF file. 
3. Handwrite on the first page, e.g. with the stylus, write "ABC".
4. Save and exit the editor. (Press the save icon 💾 in the upper left, then the "left arrow" that appears in its place once saving is done.) You should be on the "Recent notes" screen now.
5. Open the note again, and handwrite something else on the first page, e.g. "DEF".
6. Repeat step 4.
7. Open the note again. The first page will be fine. But if you scroll down to the second page, instead of seeing the contents of the second page of the PDF, you'll see the dreaded "loading circle" ! 

**Saber version**: v0.19.2 F-Droid (19020)
**Device**: Galaxy Tab A with S Pen (SM-P200)
**OS**: Android 11

**Screenshot:** ![Screenshot_20240210-140410](https://github.com/saber-notes/saber/assets/5407450/fb88f2f5-0ba5-4903-8038-88e098244706)

P.S. This is an amazing app! 🙏 Keep up the good work!
